File: ImportPrompts.cs
using System;
using System.IO;
using MongoDB.Bson;
using MongoDB.Driver;
using Newtonsoft.Json;

class ImportPrompts
{
    static async System.Threading.Tasks.Task Main(string[] args)
    {
        try
        {
            var mongoUri = args.Length > 0
                ? args[0]
                : Environment.GetEnvironmentVariable("PROMPT_PRESET_MONGO_URI")
                    ?? Environment.GetEnvironmentVariable("MONGODB_URI")
                    ?? "mongodb://localhost:27017";
            var dbName = args.Length > 1
                ? args[1]
                : Environment.GetEnvironmentVariable("PROMPT_PRESET_MONGO_DB")
                    ?? Environment.GetEnvironmentVariable("MONGODB_DB")
                    ?? "TaskPaneAddin";
            var collectionName = args.Length > 2
                ? args[2]
                : Environment.GetEnvironmentVariable("PROMPT_PRESET_MONGO_COLL")
                    ?? "PromptPresetCollection";

            var mongoClient = new MongoClient(mongoUri);
            var database = mongoClient.GetDatabase(dbName);
            var collection = database.GetCollection<BsonDocument>(collectionName);

            // Read refactored prompts
            string jsonPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "RefactoredPrompts.json");
            string jsonContent = File.ReadAllText(jsonPath);
            var prompts = JsonConvert.DeserializeObject<dynamic>(jsonContent);

            // Delete existing prompts
            var deleteResult = await collection.DeleteManyAsync(Builders<BsonDocument>.Filter.Empty);
            Console.WriteLine($"✓ Deleted {deleteResult.DeletedCount} existing prompts");

            // Insert new prompts
            var documents = new System.Collections.Generic.List<BsonDocument>();
            foreach (var prompt in prompts)
            {
                documents.Add(BsonDocument.Parse(JsonConvert.SerializeObject(prompt)));
            }

            await collection.InsertManyAsync(documents);
            Console.WriteLine($"✓ Inserted {documents.Count} new refactored prompts");

            // Verify
            var count = await collection.CountDocumentsAsync(Builders<BsonDocument>.Filter.Empty);
            Console.WriteLine($"✓ Total prompts in collection: {count}");
            Console.WriteLine("\n✓ MongoDB PromptPresetCollection successfully updated!");
        }
        catch (Exception ex)
        {
            Console.Error.WriteLine($"✗ Error: {ex.Message}");
            System.Environment.Exit(1);
        }
    }
}

File: ImportPrompts.js
const { MongoClient } = require('mongodb');
const fs = require('fs');
const path = require('path');

const MONGO_URI = 'mongodb://localhost:27017';
const DB_NAME = 'TaskPaneAddin';
const COLLECTION_NAME = 'PromptPresetCollection';

async function importPrompts() {
  const client = new MongoClient(MONGO_URI);

  try {
    await client.connect();
    console.log('Connected to MongoDB');

    const db = client.db(DB_NAME);
    const collection = db.collection(COLLECTION_NAME);

    // Read the refactored prompts
    const filePath = path.join(__dirname, 'RefactoredPrompts.json');
    const data = fs.readFileSync(filePath, 'utf8');
    const prompts = JSON.parse(data);

    // Delete existing documents
    const deleteResult = await collection.deleteMany({});
    console.log(`Deleted ${deleteResult.deletedCount} existing prompts`);

    // Insert new prompts
    const insertResult = await collection.insertMany(prompts);
    console.log(`Inserted ${insertResult.insertedCount} new prompts`);

    // Verify
    const count = await collection.countDocuments({});
    console.log(`Total prompts in collection: ${count}`);

    // Show inserted IDs
    console.log('Inserted document IDs:', Object.keys(insertResult.insertedIds).map(i => insertResult.insertedIds[i]));

  } catch (error) {
    console.error('Error:', error.message);
    process.exit(1);
  } finally {
    await client.close();
    console.log('Connection closed');
  }
}

importPrompts();
